include/pybind11/typing.h

```diff
@@ -131,6 +131,13 @@ struct handle_type_name<typing::Tuple<Types...>> {
     static constexpr auto name = const_name("tuple[")
                                  + ::pybind11::detail::concat(make_caster<Types>::name...)
                                  + const_name("]");
+    static constexpr auto arg_name
```

Comment:
How many of the added `arg_name` and `return_name` are covered by unit tests?

Reply:
Currently just for 3 classes (+ one test for list of lists) but I can add tests for all.
I added an overview of implemented tests vs changes classes in my top comment (under Possible TODOs).
Should I keep adding all those tests to the filesystem test function or split them off into their own test function?

Reply:
Hm ... up to you.

Splitting off sounds best, if that's not too much trouble.